### PR TITLE
[plugin.video.catchuptvandmore] 0.2.0

### PR DESCRIPTION
**Pull request by SylvainCecchetto**
### Description

[Version 0.2.O]
[Framework] Switch to CodeQuick framework
[New] TV Guide FR
[Add Channel] STV (UK)
[Improvements] Allocine - Add new contents / fix thumb in certain cases (FR)
[Improvements] FranceTV - Add subtitle for Replay (FR)
[Add Channel] J-One (FR)
[Add Live TV] viàATV, vià93 et viàGrandParis (FR)
[Add Channel] Atresplayer (ES) - Kodi 17.6 and inputstream.adaptive "enable" are required
[Improvements] MYTF1 - Add notification when the video is not available - Get list of programs more accurate (FR)
[Fix] Taratata - Video from FranceTV (FR)
[Add Live TV] نسمة تي في (TN)
[Remove] Heart TV - Live TV (UK)
[Add Live TV] Kerrang, Magic, Kiss, The Box, Box Hits, Box Upfront (UK)
[Fix] BVN - Fix dash behind every video title - Thanks Forage (WO)
[Fix] Arte TV - Replay (WO)
[Add Channel - Live TV] La7 (IT)
[Readd Channel] TéléMB (BE)
[Fix] Canal+ - Fix some errors on Replay (FR)
[Fix] RMC Story - Dates on Replay (FR)
[Add Live TV] All Live Rai (Rai 1, 2, News 24, Sport, ...) (IT)
[Add Channel] Jack (mycanal) (FR)
[Improvements] BVN - Add time on video title - Thanks Forage (WO)
[Add Channel] Antenne Réunion (FR)
[Add Channel] Caledonia (FR)
[Add Channel] La7d (IT)
[Add Website] Les Argonautes (CA)
[Add Channel] Rai Play (IT)
[Fix] RTL 2 and Fun Radio Live TV (FR)
[Add Channel - Live TV] Tébéo (FR)
[Fix] Autoplus - website (FR)
[Change] Dailymotion content rely on youtube-dl
[Improvements] FranceTV - Replay - Add program Name in recent videos (FR)
[Fix] BFM TV / BFM Business / BFM Paris - Live TV (FR)
[Improvements] Add TV Guide (BE)
[Improvements] Add settings to activate subtitle if present in the content (desactivate by default)
[Fix] Navigation Mycanal replay / Add notification for DRM video for mycanal replay (FR)
[Remove] Specific Code for Kodi Jarvis (not supported by codequick)
[Change] Update screenshots
[Fix] Fir timezone error (issue #102)
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by Rechi** [comment]
You don't need to execute the files from a terminal (as it won't work anyway because of the xmbc python imports).
`chmod -R -x -- plugin.video.catchuptvandmore` to fix it, if you are using linux or osx.

**Comment by SylvainCecchetto** [comment]
kodi-addon-checker crash but I do not see any red error in the log?

File: plugin.video.catchuptvandmore/resources/lib/websites/lesargonautes.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                                                                                                                                                    download_mode=True)
+                                                                                                                                    
+                                                                                                                                    item.set_callback(
+                                                                                                                                                      get_video_url,
+                                                                                                                                                      item_id=item_id,
+                                                                                                                                                      video_id=video_id)
+                                                                                                                                        yield item
```
Comment (Rechi) [inline plugin.video.catchuptvandmore/resources/lib/websites/lesargonautes.py]: Indentation in the above two functions is completely wrong, this isn't valid python.

Comment (SylvainCecchetto) [reply plugin.video.catchuptvandmore/resources/lib/websites/lesargonautes.py]: Oups, thank you.
Correction done

File: plugin.video.catchuptvandmore/resources/lib/channels/tv_guides/fr_live.py [hunk 2]
```diff
@@ -0,0 +1,683 @@
...
+            url,
+            headers={'User-agent': self._API_USER_AGENT},
+            max_age=-1)
+
+        '''
+        with requests.Session() as session:
+            response = session.get(
+                url, headers={'User-agent': self._API_USER_AGENT})
+            return response
+        '''
```
[the remark this answers is not in the document]
Comment (SylvainCecchetto) [reply plugin.video.catchuptvandmore/resources/lib/channels/tv_guides/fr_live.py]: Done

File: plugin.video.catchuptvandmore/resources/lib/labels.py [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+from __future__ import unicode_literals
+from codequick import Script, utils
+
+"""
+
+LABELS dict is only used to
+retrieve correct element in strings.po.
+If the item has no translation just use a string.
+"""
+LABELS = {
```
[the remark this answers is not in the document]
Comment (SylvainCecchetto) [reply plugin.video.catchuptvandmore/resources/lib/labels.py]: Done

**Comment by SylvainCecchetto** [comment]
Hi everyone!
Is our plugin still in test on your side?
Thank you for your work!

**Comment by wwark** [comment]
Hi Everyone,
Do you have time to finish the validation of this PR?

Thanks for your help !
Wwark.

**Comment by SylvainCecchetto** [comment]
Hi,
Our plugin is now in version [0.2.5](https://github.com/Catch-up-TV-and-More/plugin.video.catchuptvandmore/releases/tag/v0.2.5) with a lot of fixes and a new feature concerning Live TV channels.
Do you think that is a good idea to close this PR in order to open a new one with the last version?
Because this PR is opens since 4 months now :-/

Thank you for your time!

**Comment by enen92** [comment]
@SylvainCecchetto please submit the most recent version. I'll close this PR for now.

**Pull request closed without merging**